server/server.go

```diff
@@ -2086,7 +2086,13 @@ func (s *Server) checkPathScope(ctx context.Context, txn storage.Transaction, pa
 	spathParts := strings.Split(spath, "/")
 
 	for name, roots := range bundleRoots {
+		if roots == nil {
+			return types.BadRequestErr(fmt.Sprintf("all paths owned by bundle %q", name))
+		}
 		for _, root := range roots {
+			if root == "" {
+				return types.BadRequestErr(fmt.Sprintf("all paths owned by bundle %q", name))
+			}
 			if isPathOwned(spathParts, strings.Split(root, "/")) {
```

Comment:
Nit: It would be nice if `isPathOwned` contained all the root checking logic in one place.

Reply:
Agreed. I wanted the error messages to be slightly different though depending on which case was hit. I could probably have `isPathOwned` return `types.BadRequestErr` directly, but it didn't feel right when trying it out 😄